### PR TITLE
Replace symbol: factor address_of_aware variant to use the unchecked variant

### PR DESCRIPTION
**Pull request by smowton**
Just a simple refactor that proved useful out-of-tree to avoid duplicating code.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] My contribution is formatted in line with CODING_STANDARD.md.
- ~Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.~
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- ~My commit message includes data points confirming performance improvements (if claimed).~
- [x] My PR is restricted to a single feature or bugfix.
- ~White-space or formatting changes outside the feature-related changed lines are in commits of their own.~

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 0c040c68cc4ec83ff79d2bc4c6ddcf29748cd7d4).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87650389

**Review by owen-mc-diffblue — APPROVED** [review]

File: src/util/replace_symbol.cpp [hunk 1]
```diff
@@ -323,17 +323,14 @@ bool address_of_aware_replace_symbolt::replace(exprt &dest) const
...
     return true;
 
-  const exprt &e = it->second;
-
-  if(require_lvalue && !is_lvalue(e))
+  if(require_lvalue && !is_lvalue(s_copy))
     return true;
 
-  static_cast<exprt &>(s) = e;
+  static_cast<exprt &>(s) = s_copy;
```
Comment (tautschnig) [inline src/util/replace_symbol.cpp]: `s = s_copy;` now actually works, no need for a cast anymore.

**Review by tautschnig — APPROVED** [review]
Ok, but see one further change below that IMHO should be included.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 3d03d3246aba68b4828297c063d223cc9f775e76).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87739201

**Pull request merged**